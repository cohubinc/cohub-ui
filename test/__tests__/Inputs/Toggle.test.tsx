import React from "react";
import { mount } from "enzyme";
import { render, configure, fireEvent } from "@testing-library/react";

import { Inputs, StoryHelpers } from "dist";

configure({ testIdAttribute: "data-qa" });

describe("Toggle Input", () => {
  it("renders checked without crashing", async () => {
    const wrapper = mount(
      <Inputs.Toggle
        input={{
          name: "toggle-input",
          onBlur: () => ({} as any),
          onFocus: () => ({} as any),
          onChange: () => ({} as any),
          value: true
        }}
        meta={{}}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it("renders unchecked without crashing", async () => {
    const wrapper = mount(
      <Inputs.Toggle
        input={{
          name: "toggle-input",
          onBlur: () => ({} as any),
          onFocus: () => ({} as any),
          onChange: () => ({} as any),
          value: false
        }}
        meta={{}}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it("toggles on click", () => {
    const { getByTestId, getByLabelText } = render(
      <StoryHelpers.StateContainer defaultState={false}>
        {({ state, setState }) => (
          <Inputs.Toggle
            data-qa="toggleInput"
            label="Toggle Test Label"
            input={{
              value: state,
              onChange: setState
            }}
          />
        )}
      </StoryHelpers.StateContainer>
    );

    expect(getByTestId("toggleInput")).toHaveAttribute("data-checked", "false");

    fireEvent.click(getByLabelText("Toggle Test Label"));

    expect(getByTestId("toggleInput")).toHaveAttribute("data-checked", "true");
  });
});
