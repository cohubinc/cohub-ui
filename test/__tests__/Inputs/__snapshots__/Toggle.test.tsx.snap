// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toggle Input renders checked without crashing 1`] = `
<Toggle$1
  input={
    Object {
      "name": "toggle-input",
      "onBlur": [Function],
      "onChange": [Function],
      "onFocus": [Function],
      "value": true,
    }
  }
  meta={Object {}}
>
  <div
    className=" Toggle-module_labelContainerRight__3t8zx cursor-pointer "
    data-checked={true}
    data-qa="toggle"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    tabIndex={0}
  >
    <div
      className="Toggle-module_containerActive__1jtDw"
    >
      <div
        className="Toggle-module_toggleActive__33s_R"
      />
    </div>
  </div>
</Toggle$1>
`;

exports[`Toggle Input renders unchecked without crashing 1`] = `
<Toggle$1
  input={
    Object {
      "name": "toggle-input",
      "onBlur": [Function],
      "onChange": [Function],
      "onFocus": [Function],
      "value": false,
    }
  }
  meta={Object {}}
>
  <div
    className=" Toggle-module_labelContainerRight__3t8zx cursor-pointer "
    data-checked={false}
    data-qa="toggle"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    tabIndex={0}
  >
    <div
      className="Toggle-module_containerInactive__3RAMH"
    >
      <div
        className="Toggle-module_toggleInactive__1lJfx"
      />
    </div>
  </div>
</Toggle$1>
`;
